docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md

```diff
@@ -0,0 +1,204 @@
+
+ - Git repository structure clean-up
+ - Feature activation
+ - Setup default revision and few Project Settings
+
+#### Git repository structure clean-up
+
+With the new Workflow we have decided to change and unify the Git repository structure; prior to this change we were supporting two different folder structures based on whether the Project is using [`kustomize`](/console/project-configuration/kustomize-your-configurations/migrate-to-kustomize.md); now we are supporting a single folder structure that contains:
+
+- an `environments` directory, automatically generated by the Console, containing one directory for each environment, inside which all the necessary configurations are generated and stored after each deploy;
+- a `configuration` directory, with all your global customizations to be applied to all environments, and a `kustomization.yaml` manifest if you are using [`Kustomize`](/console/project-configuration/kustomize-your-configurations/manage-a-kustomize-project);
+- an `overlays` directory, containing one directory for each environment, inside which all your environment-specific patch files are stored, as well as a `kustomization.yaml` manifest if you are using [`Kustomize`](/console/project-configuration/kustomize-your-configurations/manage-a-kustomize-project).
+
+The final repository structure should be similar to the following: 
+
+```
+├── overlays
+│ ├── dev
+│ │ └─── <dev custom file>
+│ ├── production
+│ │ └─── <prod custom file>
+│ ├── other_env_id
+│ │ └─── <env_id custom file>
+│ └── ....
+│
+├── configuration
+│ └─── <global custom files>
+│
+├── environments
+│ ├── dev
+│ │ └── <k8s config files>
+│ ├── production
+│ │ └── <k8s config files>
+│ ├── other_env_id
+│ │ └── <k8s config files>
+│ └ .....
+```
+
+:::info
+All the contents of the `environments` directory are generated from scratch each time, any additional file that is not managed by the Console will be overwritten!
+:::
+
+:::caution
+If you are working on a Project with Kustomize, ensure that the `overlays/:env/kustomization.yml` file imports the proper resource. 
+To do this, replace the path `../../configuration` with `../../environments/<environment-id>` within the `environments/:env/kustomization.yml` instead of the `configuration/kustomization.yml`.
+When generating the `environments/:env/kustomization.yml`, the Console will automatically import the `configuration/kustomization.yml` if present.
+
+As such the final look of the `kustomization.yaml` within the overlays directory should look like the following example:
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ../../environments/dev
+```
+:::
+
+#### Feature Activation
+
+Follow up what's explained in the [Activating the feature](#activating-the-feature) section, if you are activating the feature on a single Project proceed with enabling the `ENABLE_CONF_GENERATION_ON_DEPLOY` feature toggle on the specific Project rather than on the whole Company.
```

Comment:
```suggestion
To activate the feature, follow the instructions in the [dedicated section](#activating-the-feature). If you wish to activate the feature on a single Project, proceed with enabling the `ENABLE_CONF_GENERATION_ON_DEPLOY` feature toggle on the specific Project rather than on the whole Company.
```